_episodes/03-high-dim-modeling.md

```diff
@@ -4,43 +4,121 @@ teaching: 0
+
+## Training data 
+* The text below is borrowed form this lesson: https://carpentries-incubator.github.io/machine-learning-novice-sklearn/01-introduction/index.html)
+* 
+Many (but not all) machine learning systems “learn” by taking a series of input data and output data and using it to form a model. The maths behind the machine learning doesn’t care what the data is as long as it can represented numerically or categorised.
+
+Typically we will need to train our models with hundreds, thousands or even millions of examples before they work well enough to do any useful predictions or classifications with them.
+
+Some systems will do training as a one shot process which produces a model. Others might try to continuosly refine their training through the real use of the system and human feedback to it. For example every time you mark an email as spam or not spam you are probably contributing to further training of your spam filter’s model.
+
+## **The curse of dimensionality**
+- The phrase, attributed to Richard Bellman, was coined to express the difficulty of using brute force (a.k.a. grid search) to optimize a function with too many input variables.
+- In machine learning, the Curse of Dimensionality refers to a set of problems that arise when working with high-dimensional data.
+  - Compute Time: Generally speaking, as the number of features in a model increases, so does the amount of time required to opitmize the model
+  - Overfitting: When there are more features than training observations, overfitting is gauranteed
+  - Interpretability: Models become more difficult to interpret as the numbers of features increase
+  - Clustering: Too many dimensions causes every observation in your dataset to appear equidistant from all the others — making it difficult to form clusters on the data. If the distances are all approximately equal, then all the observations appear equally alike (as well as equally different), and no meaningful clusters can be formed.
+
+### **Bias-Variance Tradeoff (Underftting vs Overfitting)**
+- Train error:
+- Test error: 
+- Overfitting: 
+
+**Bias**: Bias is the model's average prediction error (i.e. model's predicted values minus actual values). A model with high bias pays very little attention to the training data and oversimplifies the model. It leads to a high error on both the train and test datasets.
+
+**Variance**: Variance is the variability of the model's predictions. It indicates the overal spread of the model's predictions. A model with high variance pays too much attention to the training data (i.e. overfits) 
+
+10X as many datapoints as features as general rule of thumb
+![image alt text](https://upload.wikimedia.org/wikipedia/commons/9/9f/Bias_and_variance_contributing_to_total_error.svg)
+![need to find equivalent image w/ creative commons license](https://www.cs.cornell.edu/courses/cs4780/2018fa/lectures/images/bias_variance/bullseye.png)
```

Comment:
We shouldn't include this if it doesn't have a license that allows for reuse - You could maybe still leave it in a note instead of rendering it in the page.